File: ruby/clock/clock.rb
class Clock
  attr_reader :time

  def initialize(hour, minutes)
    @time = Time.utc(2015, 01, 01, hour, minutes)
  end

  def self.at(hour, minutes = 0)
    new(hour, minutes)
  end

  def self.from(time)
    new(time.hour, time.min)
  end

  def +(minutes)
    self.class.from(time + (minutes * 60))
  end

  def -(minutes)
    self.class.from(time - (minutes * 60))
  end

  def ==(other)
    to_s == other.to_s
  end

  def to_s
    time.strftime('%H:%M')
  end
end

File: javascript/clock/clock.js
function at(hours, minutes) {
  var min = 1000 * 60;
  var hr = min * 60;

  var Clock = {};
  var value = (~~hours * hr) + (~~minutes * min);

  Clock.valueOf = function () {
    return value;
  };

  Clock.toString = function () {
    var time = new Date(value).toISOString().split('T')[1].split(':');
    return time[0] + ':' + time[1];
  };

  Clock.plus = function (minutes) {
    value += ~~minutes * min;
    return Clock;
  };

  Clock.minus = function (minutes) {
    value -= ~~minutes * min;
    return Clock;
  };

  Clock.equals = function (other) {
    return +Clock === +other;
  };

  return Clock;
};

module.exports.at = at;